prow/gerrit/reporter/reporter.go

```diff
@@ -18,26 +18,80 @@ limitations under the License.
 package reporter
 
 import (
+	"errors"
+	"fmt"
+	"strings"
+
 	"github.com/sirupsen/logrus"
 
 	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/gerrit/client"
 )
 
-// Client holds a gerrit client
+type gerritClient interface {
+	SetReview(instance, id, revision, message string) error
+}
+
+// Client is a gerrit reporter client
 type Client struct {
-	// TODO(krzyzacy): fill this up
-	// gc gerrit.Client
+	gc gerritClient
 }
 
 // NewReporter returns a reporter client
-func NewReporter() *Client {
-	return &Client{}
+func NewReporter(cookiefilePath string, projects map[string][]string) (*Client, error) {
+	gc, err := gerrit.NewClient(projects)
+	if err != nil {
+		return nil, err
+	}
+	gc.Start(cookiefilePath)
+	return &Client{
+		gc: gc,
+	}, nil
+}
+
+// GetName returns the name of the reporter
+func (c *Client) GetName() string {
+	return "gerrit-reporter"
+}
+
+// ShouldReport returns if this prowjob should be reported by the gerrit reporter
+func (c *Client) ShouldReport(pj *v1.ProwJob) bool {
```

Comment:
Honor skipReport == true?

Reply:
that's handled in https://github.com/kubernetes/test-infra/pull/9380/files#diff-0e4877bfa35d63bade3a386f20923299R212

(it will be respected for all reporters)